backend/src/main/java/com/javahelp/backend/domain/user/register/UserRegisterInteractor.java

```diff
@@ -0,0 +1,126 @@
+package com.javahelp.backend.domain.user.register;
+
+import com.javahelp.backend.data.IUserStore;
+import com.javahelp.model.user.User;
+import com.javahelp.model.user.UserInfo;
+import com.javahelp.model.user.UserPassword;
+
+/**
+ * An interactor to make new {@link User} and {@link UserPassword} for all types of users
+ */
+public abstract class UserRegisterInteractor <T extends IUserRegisterInputBoundary,
+        U extends UserInfo>{
+
+    IUserStore userStore;
+
+    /**
+     * Creates a new {@link UserRegisterInteractor} instance
+     *
+     * @param userStore  {@link IUserStore} to use
+     */
+    public UserRegisterInteractor(IUserStore userStore) {
+        this.userStore = userStore;
+    }
+
+    /**
+     * Generate a new pseudo {@link User}. Private because this is exclusively for a general
+     * {@link User} type.
+     *
+     * @return a new {@link User} entity
+     */
+    private User register(T boundary){
+        return new User("", this.bundleInfo(boundary), boundary.getUsername());
```

Comment:
There is no need for a method that just calls the `User` constructor. This can just be a line of code, it is only called once, and does not really add any functionality or readability IMHO.